TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/UserService.php

```diff
@@ -214,7 +214,25 @@ public function getCurrentUser()
         if ($this->securityContext->canBeInitialized() === true) {
             $account = $this->securityContext->getAccount();
             if ($account !== null) {
-                return $this->getUser($account->getAccountIdentifier());
+                return $this->getUser($account->getAccountIdentifier(), $account->getAuthenticationProviderName());
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns the username of the currently logged in user, if any
+     *
+     * @return string The username or null
+     * @api
+     */
+    public function getCurrentUsername()
```

Comment:
Is this related?

Reply:
I'm still skeptical about this one: This is the UserService for Neos Users, but that method might return an unrelated Account identifier